### PR TITLE
Fix Count(*) on empty relation returns NULL when optimize_mixed_distinct_aggregation is turned on.

### PR DESCRIPTION
**Pull request by kaka11chen**
Introduce coalesce(x, 0) on a projection node to fix Count(*) on empty relation returns NULL when optimize_mixed_distinct_aggregation is turned on.

### CODE REVIEW

**Comment by kaka11chen** [comment]
@sopel39 Could you please review it, thanks.

File: presto-main/src/main/java/com/facebook/presto/sql/planner/optimizations/OptimizeMixedDistinctAggregations.java [hunk 1]
```diff
@@ -183,6 +200,23 @@ public PlanNode visitAggregation(AggregationNode node, RewriteContext<Optional<A
...
+            }
+
+            Assignments.Builder outputSymbols = Assignments.builder();
+            for (Symbol symbol : aggregationNode.getOutputSymbols()) {
+                if (coalesceSymbols.get(symbol) != null) {
+                    Expression expression = new CoalesceExpression(symbol.toSymbolReference(), new Cast(new LongLiteral("0"), "bigint"));
+                    outputSymbols.put(coalesceSymbols.get(symbol), expression);
+                }
+                else {
+                    outputSymbols.put(symbol, symbol.toSymbolReference());
```
Comment (sopel39) [inline presto-main/src/main/java/com/facebook/presto/sql/planner/optimizations/OptimizeMixedDistinctAggregations.java]: use: `com.facebook.presto.sql.planner.plan.Assignments.Builder#putIdentity` instead

File: presto-main/src/main/java/com/facebook/presto/sql/planner/optimizations/OptimizeMixedDistinctAggregations.java [hunk 2]
```diff
@@ -183,6 +200,23 @@ public PlanNode visitAggregation(AggregationNode node, RewriteContext<Optional<A
...
                     Optional.empty(),
                     node.getGroupIdSymbol());
+
+            if (coalesceSymbols.isEmpty()) {
+                return aggregationNode;
+            }
+
+            Assignments.Builder outputSymbols = Assignments.builder();
+            for (Symbol symbol : aggregationNode.getOutputSymbols()) {
+                if (coalesceSymbols.get(symbol) != null) {
```
Comment (sopel39) [inline presto-main/src/main/java/com/facebook/presto/sql/planner/optimizations/OptimizeMixedDistinctAggregations.java]: use `contains` instead

File: presto-main/src/main/java/com/facebook/presto/sql/planner/optimizations/OptimizeMixedDistinctAggregations.java [hunk 3]
```diff
@@ -167,14 +171,27 @@ public PlanNode visitAggregation(AggregationNode node, RewriteContext<Optional<A
...
                     QualifiedName functionName = QualifiedName.of("arbitrary");
-                    aggregations.put(entry.getKey(), new Aggregation(
-                            new FunctionCall(functionName, functionCall.getWindow(), false, ImmutableList.of(argument.toSymbolReference())),
-                            getFunctionSignature(functionName, argument),
-                            Optional.empty()));
+                    String signatureName = entry.getValue().getSignature().getName();
+                    if (signatureName.equals("count")
+                            || signatureName.equals("count_if") || signatureName.equals("approx_distinct")) {
+                        Symbol newSymbol = symbolAllocator.newSymbol("expr", symbolAllocator.getTypes().get(entry.getKey()));
+                        aggregations.put(newSymbol, new Aggregation(
```
Comment (sopel39) [inline presto-main/src/main/java/com/facebook/presto/sql/planner/optimizations/OptimizeMixedDistinctAggregations.java]: Move ```new Aggregation(
                                new FunctionCall(functionName, functionCall.getWindow(), false, ImmutableList.of(argument.toSymbolReference())),
                                getFunctionSignature(functionName, argument),
                                Optional.empty()));``` outside of if

**Comment by kaka11chen** [comment]
@sopel39, code updated. Thanks for your review.

**Comment by sopel39** [comment]
`testFinalStageInfo(com.facebook.presto.execution.TestSqlStageExecution` failed, but seems unrelated

**Comment by sopel39** [comment]
Merged, thanks!

**Pull request merged**